### PR TITLE
add proxy_pass if old airflow url has api rather than redirect

### PR DESCRIPTION
**Pull request by andriisoldatenko**
tl;dr Fix without adding flag

**it works without follow redirect for old links:**

```
 curl -X GET 'https://heliocentric-spectrum-1721-airflow.staging.astronomer.io/api/experimental/dags/example_dag/dag_runs' -H 'cookie: astronomer_staging_astronomer_io_auth=token'

   {
    "dag_id": "example_dag",
    "dag_run_url": "/heliocentric-spectrum-1721/airflow/graph?dag_id=example_dag&execution_date=2020-03-17+12%3A21%3A53%2B00%3A00",
    "execution_date": "2020-03-17T12:21:53+00:00",
    "id": 7,
    "run_id": "manual__2020-03-17T12:21:53+00:00",
    "start_date": "2020-03-17T12:21:53.828586+00:00",
    "state": "running"
  },
  {
    "dag_id": "example_dag",
    "dag_run_url": "/heliocentric-spectrum-1721/airflow/graph?dag_id=example_dag&execution_date=2020-03-17+12%3A25%3A01%2B00%3A00",
    "execution_date": "2020-03-17T12:25:01+00:00",
    "id": 8,
    "run_id": "manual__2020-03-17T12:25:01+00:00",
    "start_date": "2020-03-17T12:25:01.909693+00:00",
    "state": "running"
  }
]



 curl -X POST 'https://heliocentric-spectrum-1721-airflow.staging.astronomer.io/api/experimental/dags/example_dag/dag_runs' -d '{}' -H 'cookie: astronomer_staging_astronomer_io_auth=token' --compressed
{"execution_date":"2020-03-17T12:50:11+00:00","message":"Created <DagRun example_dag @ 2020-03-17 12:50:11+00:00: manual__2020-03-17T12:50:11+00:00, externally triggered: True>"}
```

**New style links:**

```
curl -X GET 'https://deployments.staging.astronomer.io/heliocentric-spectrum-1721/airflow/api/experimental/dags/example_dag/dag_runs' -d '{}' -H 'cookie: astronomer_staging_astronomer_io_auth=token' --compressed

  {
    "dag_id": "example_dag",
    "dag_run_url": "/heliocentric-spectrum-1721/airflow/graph?dag_id=example_dag&execution_date=2020-03-17+12%3A50%3A11%2B00%3A00",
    "execution_date": "2020-03-17T12:50:11+00:00",
    "id": 9,
    "run_id": "manual__2020-03-17T12:50:11+00:00",
    "start_date": "2020-03-17T12:50:11.908580+00:00",
    "state": "running"
  },
  {
    "dag_id": "example_dag",
    "dag_run_url": "/heliocentric-spectrum-1721/airflow/graph?dag_id=example_dag&execution_date=2020-03-17+12%3A51%3A06%2B00%3A00",
    "execution_date": "2020-03-17T12:51:06+00:00",
    "id": 10,
    "run_id": "manual__2020-03-17T12:51:06+00:00",
    "start_date": "2020-03-17T12:51:06.073597+00:00",
    "state": "running"
  }
]

curl -X POST 'https://heliocentric-spectrum-1721-airflow.staging.astronomer.io/api/experimental/dags/example_dag/dag_runs' -d '{}' -H 'cookie: astronomer_staging_astronomer_io_auth=token' --compressed
{"execution_date":"2020-03-17T12:50:11+00:00","message":"Created <DagRun example_dag @ 2020-03-17 12:50:11+00:00: manual__2020-03-17T12:50:11+00:00, externally triggered: True>"}
```

### CODE REVIEW

**Comment by andriisoldatenko** [comment]
UI redirect works as before:
image

**Comment by andriisoldatenko** [comment]
flower also works:
image

**Comment by andriisoldatenko** [comment]
let me know if you have tests in mind.

**Comment by schnie** [comment]
Nice! Do we need a patch to redirect flower `Ingress` too?

File: templates/ingress.yaml [hunk 1]
```diff
@@ -24,6 +24,10 @@ metadata:
     nginx.ingress.kubernetes.io/auth-signin: https://app.{{ .Values.ingress.baseDomain }}/login
     nginx.ingress.kubernetes.io/auth-response-headers: authorization, username, email
     nginx.ingress.kubernetes.io/proxy-cookie-path: / /{{ .Release.Name }}/
+    nginx.ingress.kubernetes.io/server-snippet: |
```
Comment (ashb) [inline templates/ingress.yaml]:
```suggestion
    # To support people who are using Airflow's experimental API on the old URL, but don't
    # have their client set up to follow redirects correctly: We proxy it to the right location
    # for them.
    nginx.ingress.kubernetes.io/server-snippet: |
```

**Comment by andriisoldatenko** [comment]
>Nice! Do we need a patch to redirect flower Ingress too?
no we don't, new airflow api works, if somebody asks let's fix but i'll try to avoid, cause flower ingress is a bit more complicated.

File: templates/ingress.yaml (hunk 1 above)
Comment (andriisoldatenko) [reply templates/ingress.yaml]: fixed.

**Pull request merged**